src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -2764,6 +2823,58 @@ class StorageLocation(BaseSegment):
     )
 
 
+class S3StorageIntegrationParameters(BaseSegment):
+    """Parameters for an S3 Storage Integration in Snowflake.
+
+    https://docs.snowflake.com/en/sql-reference/sql/create-storage-integration.html
+    """
+
+    name = "s3_storage_integration_parameters"
+    type = "storage_integration_parameters"
+
+    match_grammar = AnySetOf(
+        Sequence("STORAGE_PROVIDER", Ref("EqualsSegment"), "S3"),
+        Sequence(
+            "STORAGE_AWS_ROLE_ARN", Ref("EqualsSegment"), Ref("QuotedLiteralSegment")
+        ),
+        Sequence(
+            "STORAGE_AWS_OBJECT_ACL",
+            Ref("EqualsSegment"),
+            Ref(
+                "QuotedLiteralSegment"
+            ),  # TODO: Can only be 'bucket-owner-full-control'.
```

Comment:
For one ofs, rather than defining a segment, you can embed a StringParser directly in the dialect:

```suggestion
            StringParser("'bucket-owner-full-control'", CodeSegment, type="literal"),
```